T4857: snmp: Fix error when not defining client|network under community

nicolas-fort opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
After including FRR recommendations on snmp, we get error when not defining client or network under community.
Error reported in this [blog post](https://forum.vyos.io/t/unable-to-query-snmp-anymore-in-a-more-recent-1-4-version/10388).

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://phabricator.vyos.net/T4857

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->
snmp

## Proposed changes
<!--- Describe your changes in detail -->

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->

```
## Simple config, with no client/network defined
vyos@vyos:~$ show config comm | grep snmp
set service snmp community FOO authorization 'ro'
set service snmp contact 'EMAIL'
set service snmp location 'LOCATION'

## Check service is running (before this patch an error was noted here)
vyos@vyos:~$ sudo systemctl status snmpd
● snmpd.service - Simple Network Management Protocol (SNMP) Daemon.
     Loaded: loaded (/lib/systemd/system/snmpd.service; disabled; vendor preset: enabled)
    Drop-In: /run/systemd/system/snmpd.service.d
             └─override.conf
     Active: active (running) since Wed 2023-02-08 13:24:31 UTC; 1min 12s ago
    Process: 3918 ExecStartPre=/bin/mkdir -p /var/run/agentx (code=exited, status=0/SUCCESS)
   Main PID: 3919 (snmpd)
      Tasks: 1 (limit: 536)
     Memory: 7.5M
        CPU: 73ms
     CGroup: /system.slice/snmpd.service
             └─3919 /usr/sbin/snmpd -LS0-5d -Lf /dev/null -u Debian-snmp -g Debian-snmp -f -p /run/snmpd.pid

Feb 08 13:24:31 vyos systemd[1]: Starting Simple Network Management Protocol (SNMP) Daemon....
Feb 08 13:24:31 vyos systemd[1]: Started Simple Network Management Protocol (SNMP) Daemon..
vyos@vyos:~$ 

## Check snmpwalk from localhost
vyos@vyos:~$ snmpwalk -v 2c -c FOO localhost
SNMPv2-MIB::sysDescr.0 = STRING: VyOS 1.4-rolling-202302080317
SNMPv2-MIB::sysObjectID.0 = OID: SNMPv2-SMI::enterprises.44641
DISMAN-EVENT-MIB::sysUpTimeInstance = Timeticks: (17050) 0:02:50.50
SNMPv2-MIB::sysContact.0 = STRING: EMAIL
...
...
# Check restrictions when trying to get routing table:
vyos@vyos:~$ snmpwalk -v 2c -c FOO localhost .1.3.6.1.2.1.4.21
IP-MIB::ip.21 = No Such Object available on this agent at this OID

```
## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

nicolas-fort committed:
T4857: Fix error when not defining client|network under community

c-po commented on data/templates/snmp/etc.snmpd.conf.j2:
```diff
@@ -78,8 +78,8 @@ agentaddress unix:/run/snmpd.socket{{ ',' ~ options | join(',') if options is vy
 {%         if comm_config.client is not vyos_defined and comm_config.network is not vyos_defined %}
-{{ comm_config.authorization }}community {{ comm }} -V RESTRICTED
-{{ comm_config.authorization }}community6 {{ comm }} -V RESTRICTED
+{{ comm_config.authorization }}community {{ comm }} 0.0.0.0/0 -V RESTRICTED
```
This is an implicit default for the user and he will not know why. Please make this more transparent in using a `<defaultValue>` in the XML node, which will then also be added to the CLI help automatically. So the userbase will know that the network will default to 0.0.0.0/0 and ::/0.

sever-sever replied:
Is it a good idea to allow all addresses by default to get SNMP queries?
Or am I confusing something?

nicolas-fort replied:
IMO it's better to explicitly allow addresses.
This PR is just for backwards compatibility, so same configuration still works as it was working before latest changes in snmp.

But if we agree that it's better not to allow all access by default, we can close this PR, and prepare new updated and correct snmp documentation

c-po replied:
Moving this to `defaultValue` only adds "documentation" to the CLI and keeps the behavior "as is"

nicolas-fort replied:
In this case, defaultValue should set two default values: 0.0.0.0/0 and ::/0
Is this supported? As far as I know, it's not supported so far

c-po replied:
Using multiple defaults is supported, see commit 011e6a2cdea8fca73f7c433de5e2d6bc22bd9c19.

c-po committed:
snmp: T4857: explicitly define default community networks 0.0.0.0/0 and ::/0

After the RESTRICTED view was introduced snmpd requires a network to be
specified. Before adding the RESTRICTED view snmpd always assumed the default
network 0.0.0.0/0.

This commit re-adds the build in default networks for IPv4 and IPv6 and
exposes it as a proper default to the CLI so the user is informed about it:

vyos@vyos# set service snmp community foooo
Possible completions:
   authorization        Authorization type (default: ro)
+  client               IP address of SNMP client allowed to contact system
+  network              Subnet of SNMP client(s) allowed to contact system (default:
                        0.0.0.0/0, ::/0)
